addons/web/static/src/core/utils/strings.js

```diff
@@ -151,22 +151,20 @@ export function odoomark(text) {
 
 /**
  * Returns a markuped version of the input text where
- * the query is highlighted using the input classes and
- * a b tag if it is part of the text
+ * the query is highlighted using the input classes
+ * if it is part of the text.
  *
  * @param {string} query
- * @param {string} text
+ * @param {string | Markup} text
  * @param {string} classes
- * @returns {string}
+ * @returns {ReturnType<markup>}
  */
 export function highlightText(query, text, classes) {
     if (!query) {
-        return odoomark(text);
+        return htmlEscape(text);
     }
     const regex = new RegExp(`(${escapeRegExp(escape(query))})+(?=(?:[^>]*<[^<]*>)*[^<>]*$)`, "ig");
-    return markup(
-        odoomark(text).toString().replaceAll(regex, `<span class="${classes}">$1</span>`)
-    );
+    return markup(htmlEscape(text).replaceAll(regex, markup`<span class="${classes}">$1</span>`));
```

Comment:
We might want to move `htmlReplace` / `htmlReplaceAll` from `mail` to `web` instead of repeating its code.

Reply:
not in 18.3

Reply:
I made the PR with the move in master at: https://github.com/odoo/odoo/pull/215145